### PR TITLE
[stable/rabbitmq-ha] Add ingress for management and customizable probes

### PR DESCRIPTION
**Pull request by coreypobrien**
Merging features from `stable/rabbitmq` to get toward single rabbit chart for #4748

### CODE REVIEW

File: stable/rabbitmq-ha/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  annotations:
+    {{- range $key, $value := .Values.ingress.annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+spec:
+  rules:
+  {{- if .Values.ingress.hostName }}
+  - host: {{ .Values.ingress.hostName }}
```
Comment (unguiculus) [inline stable/rabbitmq-ha/templates/ingress.yaml]: Make list indentation consistent. The rest of the chart seems to indent two spaces.

File: stable/rabbitmq-ha/templates/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    http:
+  {{- else }}
+  - http:
+  {{- end }}
+      paths:
+        - path: {{ default "/" .path }}
```
Comment (unguiculus) [inline stable/rabbitmq-ha/templates/ingress.yaml]: Don't use the default function. Defaults should go in to `values.yaml`. Also, it should be `.Values.path`.

**Review by unguiculus — CHANGES_REQUESTED** [review]
I've made a couple of comments, but I'd suggest you use the ingress template you get with `helm create` using Helm 2.8+.

**Comment by coreypobrien** [comment]
@unguiculus I updated the indentation and fixed that path default etc. I'm open to making this whole chart move closer to other standards in general, but I'm trying to keep this minimally different from the `rabbitmq` chart so we can work toward merging them in #4748.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**